### DISCUSSION TITLE
Path routing with root and everything below

### DISCUSSION THREAD

**Discussion by KlemensGenetec** [dotnet/yarp#1558] [Q&A] [answered]
Hello!

I have a configuration that works but would like to get some feedback how to simplify it. Basically i want to route everything from /templates and /templates/{**everything} to a single cluster. Is that possible with a single rule? Currently I have this:

```
"Templates API": {
				"ClusterId": "template-service",
				"Match": {
					"Path": "/templates/{**catch-all}"
				},
				"Transforms": [ { "PathPattern": "/{templates/{**catch-all}" } ]
			},
			"Templates API - root": {
				"ClusterId": "template-service",
				"Match": {
					"Path": "/templates"
				},
				"Transforms": [ { "PathPattern": "/checklist-templates" } ]
			},
```

Problem is the / at the end so the /{**catchall} doesn't trigger for a path without / at the end. Am I missing something obvious?

Thanks a lot!

**Comment by Tratcher** [comment, marked as answer]
Are you using ASP.NET Core 3.1? There was a bug with catch-all routing. See:
https://docs.microsoft.com/en-us/aspnet/core/fundamentals/routing?view=aspnetcore-3.1#route-template-reference

You don't need that transform on the first route, the existing path is passed through by default.

The transform you have on the second route would prevent you from being able to combine these even if the catch-all worked.

**Reply by samsp-msft** [reply]
Triage: @Tratcher - do some verification.

**Reply by KlemensGenetec** [reply]
Actually yes, i came from net 3.1 and setup the rules back then since the root didn't work as expected. I am now at net 6 and I just tested it and it's working without the second route. Didn't know that it was a bug in 3.1 so I assumed it was still the case. Thanks for the help!

Klemens
